app/src/main/java/com/anytypeio/anytype/ui/templates/TemplateSelectAdapter.kt

```diff
@@ -0,0 +1,31 @@
+package com.anytypeio.anytype.ui.templates
+
+import androidx.fragment.app.Fragment
+import androidx.viewpager2.adapter.FragmentStateAdapter
+import com.anytypeio.anytype.presentation.templates.TemplateSelectViewModel
+
+class TemplateSelectAdapter(
+    private var templateViews: List<TemplateSelectViewModel.TemplateView>,
+    fragment: Fragment
+) : FragmentStateAdapter(fragment) {
+
+    fun update(templates: List<TemplateSelectViewModel.TemplateView>) {
+        templateViews = templateViews + templates
```

Comment:
why do you add new adapter items to the current ones? shouldn't the correct items state be provided simply by view model? otherwise, this might result in unexpected behavior.

Reply:
fixed